flybywiresim/a32nx
# fix: CVR GND CTRL

## Summary of Changes

- CVR ground control automatically resets upon first engine start, as requested by IRL A320 pilot FlyingCarpet7, and is also now a momentary push to toggle.

## Screenshots (if necessary)



## References

[YouTube for making it momentary](https://www.youtube.com/watch?v=ceDKP1qnWwg).
FCOM and other docs for logic.




## Additional context

Was previously part of #4997 

This does not implement any CVR logic except the manual recording mode. Implementing any other part would not make sense as the result would not be visible to the pilot in any way.


Discord username (if different from GitHub):

## Testing instructions

With no engines running:
- Pressing and holding GND CTL button turns GND CTL on with initial press, releasing has no further effect
- Pressing and holding GND CTL button has no effect... releasing turns GND CTL off
- Press and release GND CTL to turn it on
- Press CVR TEST button and make sure you hear a beep
- Start engine one and ensure GND CTL switches off when the oil pressure rises
- Press GND CTL and ensure it cannot be enabled while the engine is running
- Shut the engine down and repeat for engine two

General check of cockpit buttons for any regression.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
